pytorch_lightning/loggers/neptune.py

```diff
@@ -46,355 +94,557 @@ class NeptuneLogger(LightningLoggerBase):
+            used when run is created.
 
     Raises:
         ImportError:
-            If required Neptune package is not installed on the device.
+            If required Neptune package in version >=0.9 is not installed on the device.
+        TypeError:
+            If configured project has not been migrated to new structure yet.
+        ValueError:
+            If argument passed to the logger's constructor is incorrect.
     """
 
-    LOGGER_JOIN_CHAR = "-"
+    LOGGER_JOIN_CHAR = "/"
+    PARAMETERS_KEY = "hyperparams"
+    ARTIFACTS_KEY = "artifacts"
 
     def __init__(
         self,
+        *,  # force users to call `NeptuneLogger` initializer with `kwargs`
         api_key: Optional[str] = None,
-        project_name: Optional[str] = None,
-        close_after_fit: Optional[bool] = True,
-        offline_mode: bool = False,
-        experiment_name: Optional[str] = None,
-        experiment_id: Optional[str] = None,
-        prefix: str = "",
-        **kwargs,
+        project: Optional[str] = None,
+        name: Optional[str] = None,
+        run: Optional["Run"] = None,
+        log_model_checkpoints: Optional[bool] = True,
+        prefix: str = "training",
+        **neptune_run_kwargs,
     ):
-        if neptune is None:
-            raise ImportError(
-                "You want to use `neptune` logger which is not installed yet,"
-                " install it with `pip install neptune-client`."
-            )
+
+        # verify if user passed proper init arguments
+        self._verify_input_arguments(api_key, project, name, run, neptune_run_kwargs)
+
         super().__init__()
-        self.api_key = api_key
-        self.project_name = project_name
-        self.offline_mode = offline_mode
-        self.close_after_fit = close_after_fit
-        self.experiment_name = experiment_name
+        self._log_model_checkpoints = log_model_checkpoints
         self._prefix = prefix
-        self._kwargs = kwargs
-        self.experiment_id = experiment_id
-        self._experiment = None
 
-        log.info(f'NeptuneLogger will work in {"offline" if self.offline_mode else "online"} mode')
+        self._run_instance = self._init_run_instance(api_key, project, name, run, neptune_run_kwargs)
 
-    def __getstate__(self):
-        state = self.__dict__.copy()
+        self._run_short_id = self.run._short_id  # skipcq: PYL-W0212
+        try:
+            self.run.wait()
+            self._run_name = self._run_instance["sys/name"].fetch()
+        except NeptuneOfflineModeFetchException:
+            self._run_name = "offline-name"
+
+    def _init_run_instance(self, api_key, project, name, run, neptune_run_kwargs) -> Run:
+        if run is not None:
+            run_instance = run
+        else:
+            try:
+                run_instance = neptune.init(
+                    project=project,
+                    api_token=api_key,
+                    name=name,
```

Comment:
You could add some metadata like `source = lightning`, so you would know how many users are using NetptuneLogger from PyTorch Lightning.

Reply:
@tchaton Thanks for the suggestion! Actually we are already doing that with this line:
`run_instance[_INTEGRATION_VERSION_KEY] = __version__`

Reply:
We're already doing it by saving information about version in `_INTEGRATION_VERSION_KEY`